lib/url.dl

```diff
@@ -0,0 +1,95 @@
+/* Bindings for the Rust URL library.
+ */
+
+extern type Url
+
+/* Parse an absolute URL from a string. */
+extern function url_parse(url: string): Result<Url, string>
+
+/* Parse a string as an URL, with this URL as the base URL.
+ * Note: a trailing slash is significant. Without it, the last
+ * path component is considered to be a “file” name to be removed
```

Comment:
maybe an example would help, I find this comment quite hard to parse, although it's probably borrowed from the Rust library.